### ISSUE TITLE
r5.rs to be brought up to 0.12 nightly syntax

### ISSUE THREAD

**Issue by viperscape** [kimhyunkang/r6.rs#3]
I noticed most of the library as written in the earlier syntax and so these changes need to be made to build on the nightly (which is the formally recommended version to be developing in rust). I may fork and do this, if I can find some time. Thanks for the cool library!

**Comment by sanxiyn** [comment]
What happened?

**Comment by kimhyunkang** [comment]
@sanxiyn I hit a compiler bug 3 months ago, and did not have enough time to find out how to fix it :(

Maybe I'll try again next week, to see if it's magically fixed!

**Comment by kimhyunkang** [comment]
Hi.

I figured out making the code work with Rust 1.0 alpha is only little easier than re-writing the code from scratch. So I did it!

The runtime is now stack-based VM, and the compiler compiles the code into a bytecode -- actually, a bytecode-like structure I wrote.

The functionality is greatly reduced after re-write, but I feel much more confident with the new design, as it's now feasible to support advanced features of scheme, such as continuation, multiple return values, or proper garbage collection.
